Remotion/Web/ClientScript/Scripts/SmartPage.ts

```diff
@@ -258,19 +321,42 @@ class SmartPage_Context
     }
   };
 
-  public SmartPage_PageRequestManager_endRequest = function (sender: unknown, args: Sys.WebForms.EndRequestEventArgs)
+  public OverrideErrorMessageForStatusCode (statusCode: number, message: string) : void {
+    let current = this._errorMessageMap.get(statusCode);
+    if (current == undefined) return;
+
+    current.errorMessage = message;
+  }
+
+  public SmartPage_PageRequestManager_endRequest = (sender: unknown, args: Sys.WebForms.EndRequestEventArgs) =>
   {
-    if (args.get_error() != undefined && (args.get_error() as any).httpStatusCode == '500')
+    if (args.get_error() != undefined)
     {
-      const errorMessage = args.get_error().message;
-      args.set_errorHandled(true);
+      let statusCode = (args.get_error() as any).httpStatusCode;
 
-      const errorBody = '<div class="SmartPageErrorBody"><div>' + errorMessage + '</div></div>';
+      let errorDeclaration = this._errorMessageMap.get(statusCode);
 
-      SmartPage_Context.Instance!.ShowMessage("SmartPageServerErrorMessage", errorBody);
+      if (errorDeclaration == undefined || statusCode == 200){
+        this.HandleEndRequestError("undefined", args.get_error().name, "An error has occured.", args, true);
+      }
+      else {
+        this.HandleEndRequestError(statusCode, errorDeclaration.shortDescription, errorDeclaration.errorMessage, args, false);
+      }
     }
   }
 
+  private HandleEndRequestError(statusCode: string, errorDescription: string, errorMessage: string, args: Sys.WebForms.EndRequestEventArgs, showBody: boolean): void
```

Comment:
I think it would be better to pass in the `SmartPage_ErrorDeclaration` here. This allows us to add new properties without needing to add new parameters to this method. `showBody`, for example, might also be a good property to add to the class instead. That way, we can handle all HTTP status codes the same way and allow overriding the behavior for those values as well.